test/base/unit/MockConnectors.scala

```diff
@@ -0,0 +1,88 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package base.unit
+
+import builders.RegistrationBuilder
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, when}
+import org.mockito.stubbing.OngoingStubbing
+import org.scalatest.{BeforeAndAfterEach, Suite}
+import org.scalatestplus.mockito.MockitoSugar
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.{
+  IncorpIdConnector,
+  SoleTraderInorpIdConnector
+}
+import uk.gov.hmrc.plasticpackagingtax.registration.models.genericregistration.{
+  IncorporationDetails,
+  SoleTraderIncorpIdCreateRequest,
+  SoleTraderIncorporationDetails
+}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait MockConnectors extends MockitoSugar with RegistrationBuilder with BeforeAndAfterEach {
+  self: Suite =>
+
+  val mockIncorpIdConnector                               = mock[IncorpIdConnector]
+  val mockSoleTraderConnector: SoleTraderInorpIdConnector = mock[SoleTraderInorpIdConnector]
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockIncorpIdConnector, mockSoleTraderConnector)
+  }
+
+  def mockGetUkCompanyDetails(
+    incorporationDetails: IncorporationDetails
+  ): OngoingStubbing[Future[IncorporationDetails]] =
+    when(mockIncorpIdConnector.getDetails(any())(any()))
+      .thenReturn(Future(incorporationDetails))
+
+  def mockGetSoleTraderDetails(
+    soleTraderDetails: SoleTraderIncorporationDetails
+  ): OngoingStubbing[Future[SoleTraderIncorporationDetails]] =
+    when(mockSoleTraderConnector.getDetails(any())(any()))
+      .thenReturn(Future(soleTraderDetails))
+
+  def mockGetSoleTraderDetailsFailure(
+    ex: Exception
+  ): OngoingStubbing[Future[SoleTraderIncorporationDetails]] =
+    when(mockSoleTraderConnector.getDetails(any())(any()))
+      .thenThrow(ex)
+
+  def mockGetUkCompanyDetailsFailure(ex: Exception): OngoingStubbing[Future[IncorporationDetails]] =
+    when(mockIncorpIdConnector.getDetails(any())(any()))
+      .thenThrow(ex)
+
+  def mockSoleTraderCreateIncorpJourneyId(redirectUrl: String): OngoingStubbing[Future[String]] =
+    when(
+      mockSoleTraderConnector.createJourney(any[SoleTraderIncorpIdCreateRequest])(any())
+    ).thenReturn(Future.successful(redirectUrl))
+
+  def mockSoleTraderCreateIncorpJourneyIdException(): OngoingStubbing[Future[String]] =
+    when(
+      mockSoleTraderConnector.createJourney(any[SoleTraderIncorpIdCreateRequest])(any())
+    ).thenThrow(new RuntimeException("error"))
+
+  def mockukCompanyCreateIncorpJourneyId(redirectUrl: String): OngoingStubbing[Future[String]] =
```

Comment:
Extremely petty one  but `U` should be captialised 😅

Reply:
done